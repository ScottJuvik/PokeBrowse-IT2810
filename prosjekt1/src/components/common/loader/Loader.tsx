import pokeball from '@assets/pokeball.svg';
import classNames from 'classnames';
import styles from './Loader.module.css';

interface LoaderProps {
    size?: string;
    text?: string;
    customImageSrc?: string;
    className?: string;
}

const Loader = ({ size = '5rem', text = 'Catching PokÃ©mon...', customImageSrc, className }: LoaderProps) => {
    const loaderWrapperClass = classNames(styles.loaderWrapper, className);

    return (
        <div data-testid="loader" aria-busy="true" aria-live="polite" role="status" className={loaderWrapperClass}>
            <img
                src={customImageSrc || pokeball}
                alt="Loading icon"
                style={{ width: size, height: size }}
                className={styles.pokeball}
            />
            <p className={styles.loadingText} aria-label="Loading text">
                {text}
            </p>
        </div>
    );
};

export default Loader;
