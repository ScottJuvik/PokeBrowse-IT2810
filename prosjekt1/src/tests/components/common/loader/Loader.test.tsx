import gengar from '@assets/gengar.png';
import Loader from '@components/common/loader/Loader';
import { render, screen } from '@testing-library/react';

describe('Loader', () => {
    const renderComponent = (customImageSrc?: string, text?: string) => {
        render(<Loader customImageSrc={customImageSrc} text={text} />);

        return {
            image: screen.getByRole('img'),
            text: screen.getByLabelText('Loading text'),
            container: screen.getByTestId('loader'),
        };
    };

    it('renders the loader with the default props', () => {
        const { image, text } = renderComponent();
        expect(image).toBeInTheDocument();
        expect(text).toBeInTheDocument();
    });

    it('applies the custom image source', () => {
        const { image } = renderComponent(gengar);
        expect(image).toHaveAttribute('src', gengar);
    });

    it('applies the custom text', () => {
        const { text } = renderComponent(undefined, 'Custom text');
        expect(text).toHaveTextContent('Custom text');
    });

    it('sets aria-busy to true while loading', () => {
        const { image } = renderComponent();
        const loaderContainer = image.parentElement;
        expect(loaderContainer).toHaveAttribute('aria-busy', 'true');
    });

    it('matches the snapshot', () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
    });
});
