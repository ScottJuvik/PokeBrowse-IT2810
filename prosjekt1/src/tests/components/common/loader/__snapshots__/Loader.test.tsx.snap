// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader > matches the snapshot 1`] = `
<div
  aria-busy="true"
  aria-live="polite"
  class="_loaderWrapper_f6abc6"
  data-testid="loader"
  role="status"
>
  <img
    alt="Loading icon"
    class="_pokeball_f6abc6"
    src="/project1/src/assets/pokeball.svg"
    style="width: 5rem; height: 5rem;"
  />
  <p
    aria-label="Loading text"
    class="_loadingText_f6abc6"
  >
    Catching Pok√©mon...
  </p>
</div>
`;
